docs/guides/nrfconnect_examples_software_update.md

```diff
@@ -3,13 +3,73 @@
 Some examples for the development kits from Nordic Semiconductor support
 over-the-air Device Firmware Upgrade.
 
-Currently, the Bluetooth LE is the only available transport for performing the
-DFU operation and it uses nRF Connect SDK's
-[Simple Management Protocol](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/device_mgmt/index.html#device-mgmt).
-The upgrade can be done either using a smartphone application or a PC command
-line tool.
+Depending on the example, the DFU can be performed using the Matter-compliant
+OTA feature in which the Matter operational network is used for downloading a
+new firmware image, or using
+[Simple Management Protocol](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/device_mgmt/index.html#device-mgmt)
+over Bluetooth LE. In the latter case, the DFU can be done either using a
+smartphone application or a PC command line tool.
 
-## Device Firmware Upgrade using smartphone
+## Devide Firmware Upgrade over Matter
+
+> **_NOTE:_** The procedure presented below requires that you have OpenThread
+> Border Router set up either in Docker or on a Raspberry Pi. Read
+> [Setup OpenThread Border Router on Raspberry Pi](openthread_border_router_pi.md)
+> to learn how to install the OTBR on a Raspberry Pi.
+
+To test the DFU over Matter, complete the following steps:
+
+1.  Navigate to the CHIP root directory.
+2.  Build OTA Provider application for Linux:
+
+        $ scripts/examples/gn_build_example.sh examples/ota-provider-app/linux out/provider chip_config_network_layer_ble=false
+
+3.  Build chip-tool for Linux:
+
+        $ scripts/examples/gn_build_example.sh examples/chip-tool out/chiptool 'chip_mdns="platform"'
+
+4.  Run OTA Provider application with _app_update.bin_ replaced with the path to
+    the new firmware image which you wish to provide to the Matter device:
+
+         $ out/provider/chip-ota-provider-app -f app_update.bin
+
+    Keep the application running and use another terminal for the remaining
+    steps.
+
+5.  Commission the OTA Provider into the Matter network using Node Id 1:
+
+        $ ./out/chiptool/chip-tool pairing onnetwork 1 20202021
+
+6.  Use the OTBR web interface to form a new Thread network using the default
+    network settings.
+7.  Commission the Matter device into the same Matter network using Node Id 2:
+
+         $ ./out/chiptool/chip-tool pairing ble-thread 2 hex:000300000f02081111111122222222051000112233445566778899aabbccddeeff01021234 20202021 3840
+
+    Note that the parameter starting with _hex:_ prefix is the Thread network's
+    Active Operational Dataset. It can be retrieved from the OTBR in case you
+    have changed the default network settings when forming the network.
+
+8.  Initiate the DFU procedure in one of the following ways:
+
+    -   If you have built the device firmware with `-DCONFIG_CHIP_LIB_SHELL=y`
```

Comment:
Maybe we could add a separate application building step and mention this option there?

Reply:
Hmm. The guide kind of assumes that you have an example already built, so I wonder if it would be consistent. Also not sure if it's worth repeating that since the build should be described in the example's README.

Reply:
Ok